### PR TITLE
Add shortcut tests

### PR DESCRIPTION
**Pull request by ahmedalsudani**
@tejohnso ready for review.

What this change adds

- check scripts exist every 4 seconds until success
- check shortcuts point to the right scripts every 4 seconds until success

The changes have been tested on Ubuntu and Windows x64

FYI @fjvallarino

### CODE REVIEW

File: shortcuts.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+  checkShortcutsNameAndTarget(ctx, version) {
+    return new Promise((res) => {
+      function curriedCheckShortcutsNameAndTarget() {
+        log.debug("Checking shortcuts");
+        checkShortcutList(version)
+          .then(res)
+          .catch(() => {
+            log.debug("Shortcut check failed");
+            ctx.timeouts.checkShortcutList = setTimeout(curriedCheckShortcutsNameAndTarget, 4000)
```
Comment (tejohnso) [inline shortcuts.js]: @ahmedalsudani These checks run after the presentation is confirmed to be running (screenshot check).  That happens after the installer run completes and links have been updated.   So I don't think we need the repeated call on failure here.  Did you see a case where it was only passing after many seconds had elapsed?

Comment (ahmedalsudani) [inline shortcuts.js]: @tejohnso you are correct, but I wrote the test following the same pattern we used for the other tests. It's unlikely, but if we change the order of the tests or process through which the Installer creates script files, we won't need to modify this test as a result.

**Comment by tejohnso** [comment]
@ahmedalsudani I don't think that a function that takes advantage of a closure counts as currying, but :+1:

**Comment by ahmedalsudani** [comment]
@tejohnso Hmm. I think you're right, but I'm merging it as is

**Pull request merged**